agent/nmap_agent.py

```diff
@@ -372,6 +372,19 @@ def _emit_fingerprints(
                 else:
                     raise ValueError(f"Incorrect ip version {version}")
 
+                if host.get("os") is not None:
+                    if host.get("os").get("osmatch", {}) is not None:
```

Comment:
`.get("osmatch", {})` can return None? there is a default return value in `get`, or `osmatch` can actually have a value of None?